File: app/src/main/java/edu/cascadia/mobas/northcreekforest/db/PhotoDao.java
package edu.cascadia.mobas.northcreekforest.db;


import android.arch.lifecycle.LiveData;

import android.provider.ContactsContract;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import edu.cascadia.mobas.northcreekforest.models.Photo;


@Dao
public interface PhotoDao {

    //create and update
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addPhoto(Photo photo);

    @Update
    void updatePhoto(Photo photo);

    @Delete
    void deletePhoto(Photo photo);

    @Query("DELETE FROM photos_table")
    void deleteAllPhotos();

    @Query("SELECT * FROM photos_table ORDER BY id DESC")
    LiveData<List<Photo>> getAllPhotos();

}

File: app/src/main/java/edu/cascadia/mobas/northcreekforest/db/PhotoDao.kt
package edu.cascadia.mobas.northcreekforest.db


import androidx.lifecycle.LiveData

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import edu.cascadia.mobas.northcreekforest.models.Photo


@Dao
interface PhotoDao {

    @get:Query("SELECT * FROM photos_table ORDER BY id DESC")
    val allPhotos: LiveData<List<Photo>>

    //create and update
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addPhoto(photo: Photo)

    @Update
    fun updatePhoto(photo: Photo)

    @Delete
    fun deletePhoto(photo: Photo)

    @Query("DELETE FROM photos_table")
    fun deleteAllPhotos()

}
